builtin/logical/pkiext/pkiext_binary/acme_test.go

```diff
@@ -181,6 +188,33 @@ func SubtestACMECertbot(t *testing.T, cluster *VaultPkiCluster) {
 	require.NoError(t, err, "got error running double revoke command")
 	require.NotEqual(t, 0, retcode, "expected non-zero retcode double revoke command result")
 
+	// Attempt to issue against a domain that doesn't match the challenge.
+	// N.B. This test only runs locally or when the nightly regression env var is provided to CI.
+	if pkiext.IsLocalOrNightlyRegression() {
+		certbotInvalidIssueCmd := []string{
+			"certbot",
+			"certonly",
+			"--no-eff-email",
+			"--email", "certbot.client@dadgarcorp.com",
+			"--agree-tos",
+			"--no-verify-ssl",
+			"--standalone",
+			"--non-interactive",
+			"--server", directory,
+			"-d", "armoncorp.com",
```

Comment:
Before calling out to this domain, do you want to sinkhole it at the DNS resolver level, so that if it gets registered, we don't accidentally send traffic to it?

```go
cluster.Dns.AddDomain("armoncorp.com")
cluster.Dns.AddRecord("armoncorp.com", "A", "127.0.0.1")
```

Reply:
Yep, good call. It'd still fail, but best not to spam someone's domain with our test requests.